.k8s/K8s-Deployment.md

```diff
@@ -0,0 +1,45 @@
+# Kubernetes Deployment
+## Deployment
+### The Tool
+To work with a K8s cluster you need to install Kubectl, a tool for controlling your K8s deployment.
+[Kubectl installation guide](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+
+### Before Deploying
+Kubernetes manifests do not support adding config files directly in the manifest via a file path. Instead we need to use
+a resource called ConfigMaps.
+
+ConfigMaps are how we pass configurations to the containers. These can be created through literals or files. The only
+issue is that these can"t be created in a manifest based off a file like we used to do with Docker Compose. Now we have
+to created them manually using `kubectl create configmap <config-map-name> --from-file=<path-to-file>`
+
+### Deploying
+The deployment manifest file contains everything to deploy our application to the Kubernetes cluster. To deploy use:
+`kubectl apply -f search-deployment.yml`
+
+To remove a deployment and all related pods and containers use:
+`kubectl delete -f search-deployment.yml`
+
+Note: The ConfigMap will not be deleted automatically because it was added to the cluster manually.
+#### Optional:
+Install MiniKube, a tool for running Kubernetes locally in a VM or the host
+    machine. Deploys a lightweight single-node k8s cluster.
+Note: Run `eval $(minikube docker-env)` to enter Minikube"s docker context
+
+When starting up [Minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/), make sure your vm-driver is the one bundled with
+VirtualBox. Dnsmasq doesn"t play nice with the standalone hyperkit.
+
+
+## Command Cheat Sheet
+- `kubectl cluster-info`: View Cluster info
+- `kubectl get` - List resources
+    - `kubectl get nodes`: Lists available Nodes in the cluster
+    - `kubectl get pods`: Lists Pods in your cluster
+    - `kubectl get deployments`: List deployments in your cluster
+- `kubectl describe` - Show detailed information about a resource
+- `kubectl logs` - Print the logs from a container in a pod
+- `kubectl exec` - Execute a command on a container in a pod
```

Comment:
pod"s should be pod's and there"s should be there's

Reply:
I need to update the formatter to exclude `.md` files